syntaxes/omg.idl.tmLanguage.json

```diff
@@ -572,11 +573,109 @@
             "name": "annotation.omg.idl",
             "patterns": [{
                 "name": "storage.type.annotation.omg.idl",
-                "match": "@\\w+(\\(.*\\)){0,1}"
+                "match": "@\\w+\\s*(\\(.*\\)){0,1}"
             }]
+        },
+        "interface_def": {
+            "name": "meta.storage.interface.omg.idl",
+            "begin": "(interface)\\s+(\\w+)\\s*\\{",
+            "end": "\\};",
+            "beginCaptures": {
+                "1": {"name": "storage.type.interface.omg.idl"},
+                "2": {"name": "entity.name.type.omg.idl"}
+            },
+            "patterns": [
+                {"include": "#interface_body"},
+                {"include": "#single_line_comments"},
+                {"include": "#block_comments"}
+            ]
+        },
+        "interface_body": {
+            "patterns": [
+                {"include": "#attr_dcl"},
+                {"include": "#attr_dcl"},
+                {"include": "#op_dcl"}
+            ]
+        },
+        "op_dcl": {
+            "name": "meta.storage.operation.omg.idl",
+            "begin": "^\\s*((\\w+(<\\d+>){0,1})|\\w+::\\w+)\\s+\n*(\\w+)\\s*\\(",
+            "end": ";",
+            "beginCaptures": {
+                "2": {
+                    "patterns": [{"include": "#op_type_spec"}]
+                },
+                "4": {"name": "entity.name.function.omg.idl"}
+            },
+            "patterns": [
+                {"include": "#raises_keyword"},
+                {"include": "#parameter_dcl"}]
+        },
+        "raises_keyword": {
+            "patterns": [{
+                "name": "keyword.control.raises.omg.idl",
+                "match": "\\b(raises|setraises|getraises)\\b"
+            }]
+        },
+        "void_type": {
+            "patterns": [{
+                "name": "storage.type.primitive.omg.idl",
+                "match": "\\b(void)\\b"
+            }]
+        },
+        "op_type_spec": {
+            "patterns": [
+                {"include": "#void_type"},
+                {"include": "#const_type"},
+                {"include": "#template_type_spec"}
+            ]
+        },
+        "parameter_dcl": {
+            "name": "meta.storage.operation.parameter.omg.idl",
+            "begin": "(in|out|inout)\\b",
```

Comment:
Is this mandatory? If you don't specify anything I think it defaults to `in`

Reply:
Hmm... first reading of the spec made it look mandatory, but I made it optional.